Sources/Utilities/Security/SecurityProtectionManager/SecurityProtectionManager.swift

```diff
@@ -0,0 +1,199 @@
+    func firstLevelSystemCheck() {
+        UIApplication.rx.didFinishLaunching
+            .mapToVoid()
+            .subscribe(onNext: { [unowned self] in
+                checkSystemChange()
+            })
+            .disposed(by: disposeBag)
+    }
+    
+    /// Starts periodic check if device is jailbroken to detect potential reverse engineering during app runtime.
+    ///
+    /// Periodic timer is restarted every time that application becomes active.
+    func initiatePeriodicSystemCheck() {
+        UIApplication.rx.didBecomeActive
+            .startWith(())
+            .filter { [unowned self] in firstLevel == .offline }
+            .subscribe(onNext: { [unowned self] in
+                startPeriodicSystemCheck()
+            })
+            .disposed(by: disposeBag)
+    }
+
+    /// Handles background app state. Adds or removes the blur based on app state and security system states.
+    ///
+    /// - Parameter window: window on which blur will be applied
+    func protectAppInBackground(window: UIWindow?) {
+        // In modified OS version, blur everything when the app becomes inactive
+        let inModified = UIApplication.rx
+            .willResignActive
+            .filter { Self.shouldAddBlur(for: .offline) }
+            .mapTo(true)
+
+        // Non modified version is secured from the attack, so keep blur only when the app goes to the background
+        let inNonModified = UIApplication.rx
+            .didEnterBackground
+            .filter { Self.shouldAddBlur(for: .online) }
+            .mapTo(false)
+
+        handleAddBlur(animated: Observable.merge(inModified, inNonModified), in: window)
+        handleRemoveBlur(in: window)
+    }
+
+    /// Checks biometrics state every time app becomes active.
+    ///
+    func biometricChangesCheck() {
+        UIApplication.rx.didBecomeActive
+            .subscribe(onNext: { [unowned self] in
+                checkBiometricsState()
+            })
+            .disposed(by: disposeBag)
+    }
+    
+}
+
+// MARK: - Private extensions
+
+private extension SecurityProtectionManager {
+    
+    var isSimulator: Bool {
+        return TARGET_OS_SIMULATOR != 0
+    }
+    
+    static func shouldAddBlur(for level: SecurityService.SystemStatus) -> Bool {
+        // Add additional custom logic to check if blur should be added
+        return SecurityService.ServiceStatus.firstLevel == level
+    }
+    
+    func startPeriodicSystemCheck() {
+        systemCheckTimer = SecurityTimer(timeInterval: systemCheckInterval)
+        systemCheckTimer?.eventHandler = { [weak self] in
+            self?.checkSystemStatus()
+        }
+        systemCheckTimer?.resume()
+    }
+
+    func checkSystemChange() {
+        guard SecurityService.ServiceStatus.firstLevel == .offline else { return }
+        // Device is Jailbroken
+        // Add custom logic for showing alert or restricting access to the application.
+    }
```

Comment:
Let's add TODO here. That way, we can raise awareness of this method if someone just copy/paste those files.
```suggestion
    func checkSystemChange() {
        guard SecurityService.ServiceStatus.firstLevel == .offline else { return }
        // Device is Jailbroken
        // TODO: Add custom logic for showing alert or restricting access to the application.
    }
```